src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -260,21 +270,22 @@ public override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarati
                             throw ExceptionUtilities.UnexpectedValue(accessor.Kind());
                     }
 
-                    var body = (SyntaxNode?)accessor.Body ?? accessor.ExpressionBody;
-                    if (body != null)
-                    {
-                        accessorsHaveImplementation = true;
-                    }
-
                     usesFieldKeyword = usesFieldKeyword || containsFieldKeyword(accessor);
                 }
             }
             else
             {
                 var body = GetArrowExpression(syntax);
-                accessorsHaveImplementation = body is object;
+                hasGetAccessorImplementation = body is object;
+                hasSetAccessorImplementation = false;
                 usesFieldKeyword = body is { } && containsFieldKeyword(body);
-                Debug.Assert(accessorsHaveImplementation); // it's not clear how this even parsed as a property if it has no accessor list and no arrow expression.
+                Debug.Assert(hasGetAccessorImplementation); // it's not clear how this even parsed as a property if it has no accessor list and no arrow expression.
```

Comment:
Since you're modifying this line anyway, consider making this comment an actual assert message.